Expose a jupyter_server pytest plugin

Zsailer opened the pull request:
This PR moves the `conftest` into the jupyter_server module and exposes an entrypoint to that module to enable extensions to use these helpful fixtures for testing. 

To use these fixtures, you simply have to add `pytest_jupyter_server` to the `pytest_plugins` list in any test suite.

echarles commented:
@Zsailer Strange: Failing for windows py 3.8 on appveyor while succeeding on github actions. I can't remember we had such cases during our sprint to make tests green a few weeks ago...?

```
self = <ProactorEventLoop running=False closed=False debug=False>, fd = 736
callback = <bound method BaseAsyncIOLoop._handle_events of <zmq.eventloop.ioloop.ZMQIOLoop object at 0x00000099447949D0>>
args = (736, 4)
    def add_writer(self, fd, callback, *args):
>       raise NotImplementedError
E       NotImplementedError
c:\miniconda37-x64\envs\test-env-38\lib\asyncio\events.py:507: NotImplementedError
```

The print message you have added shows exactly the same information: `win32 sys.version_info(major=3, minor=8, micro=0, releaselevel='final', serial=0)`

Maybe @kevin-bates can help here?

Zsailer commented:
Hm, this was really weird. I changed a few things:
1. I removed ipykernel from the `conda install` list. I don't know why, but it appears the conda version of ipykernel was being used over the source version we cloned+installed later in the CI script. 
2. I removed to "windows +Python 3.8" conditional from the plugin. This condition is already checked by the `_init_asyncio_patch` in serverapp, so it was redundant to check for conditional again. It avoids adding extra confusion to whether the patch is applied before or after the fixture is initialized...

Anyways, getting all greens now. Whoever wants to review, go ahead and squash and merge if you find no issues.

@echarles, you should be able to start using this for your jupyterlab_server branch. For an example on how to use this plugin, checkout the conftest file from `nbclassic` in [Zsailer/nbclassic#4](https://github.com/Zsailer/nbclassic/pull/4)

Zsailer commented on jupyter_server/pytest_plugin.py:
```diff
@@ -0,0 +1,218 @@
+import nbformat
+
+# This shouldn't be needed anymore, since pytest_tornasync is found in entrypoints
+# Removing to avoid race conditions.
```
```suggestion
```

Zsailer commented on jupyter_server/pytest_plugin.py:
```diff
@@ -0,0 +1,217 @@
+# We have to override the io_loop fixture with an
+# asyncio patch. This will probably be removed in
+# the future.
+print("\n\n\n{} {} \n\n\n\n".format(sys.platform, sys.version_info))
```
```suggestion
```

echarles commented:
LGTM. Merging now.

Thx @Zsailer, this was not an easy one. I will start using the plugin for the jupyterlab_server tests.
